Source/Scene/PolygonClippingAccelerationGrid.js

```diff
@@ -0,0 +1,505 @@
+import defaultValue from "../Core/defaultValue.js";
+import Cartesian2 from "../Core/Cartesian2.js";
+import DeveloperError from "../Core/DeveloperError.js";
+
+// https://stackoverflow.com/a/2049593
+function sign(ax, ay, bx, by, cx, cy) {
+  return (ax - cx) * (by - cy) - (bx - cx) * (ay - cy);
+}
+
+function pointInTriangle(px, py, ax, ay, bx, by, cx, cy) {
+  var d1 = sign(px, py, ax, ay, bx, by);
+  var d2 = sign(px, py, bx, by, cx, cy);
+  var d3 = sign(px, py, cx, cy, ax, ay);
+  var isNegative = d1 < 0 || d2 < 0 || d3 < 0;
+  var isPositive = d1 > 0 || d2 > 0 || d3 > 0;
+  return !(isNegative && isPositive);
+}
+
+function pointInsideRect(rect, px, py) {
+  var leftWall = px <= rect.topRight.x;
+  var rightWall = px >= rect.topLeft.x;
+  var topWall = py <= rect.topLeft.y;
+  var btmWall = py >= rect.btmRight.y;
+  return leftWall && rightWall && topWall && btmWall;
+}
+
+// https://stackoverflow.com/a/58657254
+export function lineIntersection(s1x, s1y, t1x, t1y, s2x, s2y, t2x, t2y) {
+  var dX = t1x - s1x;
+  var dY = t1y - s1y;
+
+  var determinant = dX * (t2y - s2y) - (t2x - s2x) * dY;
+  if (determinant === 0) return null; // parallel lines
+
+  var lambda =
+    ((t2y - s2y) * (t2x - s1x) + (s2x - t2x) * (t2y - s1y)) / determinant;
+
+  var gamma = ((s1y - t1y) * (t2x - s1x) + dX * (t2y - s1y)) / determinant;
+
+  // check if there is an lineIntersection
+  if (!(0 <= lambda && lambda <= 1) || !(0 <= gamma && gamma <= 1)) return null;
+
+  return {
+    x: s1x + lambda * dX,
+    y: s1y + lambda * dY,
+  };
+}
+
+function doesTriangleIntersectRect(rect, ax, ay, bx, by, cx, cy) {
+  // check if the points are inside the rectangle
+  var leftWall = ax <= rect.topRight.x;
+  var rightWall = ax >= rect.topLeft.x;
+  var topWall = ay <= rect.topLeft.y;
+  var btmWall = ay >= rect.btmRight.y;
+
+  if (leftWall && rightWall && topWall && btmWall) {
+    return true;
+  }
+
+  leftWall = bx <= rect.topRight.x;
+  rightWall = bx >= rect.topLeft.x;
+  topWall = by <= rect.topLeft.y;
+  btmWall = by >= rect.btmRight.y;
+
+  if (leftWall && rightWall && topWall && btmWall) {
+    return true;
+  }
+
+  leftWall = cx <= rect.topRight.x;
+  rightWall = cx >= rect.topLeft.x;
+  topWall = cy <= rect.topLeft.y;
+  btmWall = cy >= rect.btmRight.y;
+
+  if (leftWall && rightWall && topWall && btmWall) {
+    return true;
+  }
+
+  if (
+    lineIntersection(
+      rect.topLeft.x,
+      rect.topLeft.y,
+      rect.topRight.x,
+      rect.topRight.y,
+      ax,
+      ay,
+      bx,
+      by
+    ) !== null
+  )
+    return true;
```

Comment:
Since prettier really expands the code here it would be clearer to use curly braces.

```
  if (
    lineIntersection(
      rect.topLeft.x,
      rect.topLeft.y,
      rect.topRight.x,
      rect.topRight.y,
      ax,
      ay,
      bx,
      by
    ) !== null
  ) {
    return true;
  }
```